cli/tools/registry/mod.rs

```diff
@@ -107,6 +107,19 @@ pub async fn publish(
       }
     }
   }
+
+  if let Some(version) = &publish_flags.override_version {
+    publish_configs = publish_configs
+      .into_iter()
+      .map(|mut config| {
+        let mut config_file = config.config_file.as_ref().clone();
+        config_file.json.version = Some(version.clone());
+        config.config_file = Arc::new(config_file);
+        config
+      })
+      .collect();
```

Comment:
This will set the same version for all packages when run in a workspace.

When publish with `--set-version` is run in a workspace, we have two options:
1. throw an error that you can only use the flag for individual packages
2. make `--set-version` accept `key=value` pairs as value to set the appropriate version for a package

@dsherret what do you think?

Reply:
Discussed and in a workspace we should error when this occurs for multiple packages in order to prevent accidentally publishing the entire workspace at the same version.

Reply:
We decided to go with option 1.